### PR TITLE
Add support for implicitly rendering renderables

### PR DESCRIPTION
**Pull request by joelhawksley**
### Motivation / Background

In #36388, we added support for rendering objects that respond to for controller actions. This PR adds support for implicitly rendering renderables for controller actions as one can for regular templates today.

### Detail

Specifically, I've updated `ImplictRender` to look for a corresponding renderable in the format `Prefixes::ActionNameView`.

If this approach is deemed acceptable, I'll gladly add docs changes to this PR ❤️ 

### Additional information

@bradgessler's post on doing this for Phlex: https://fly.io/ruby-dispatch/hacking-rails-implicit-rendering-for-view-components/
ViewComponent thread discussing the topic: https://github.com/ViewComponent/view_component/issues/618

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

cc @joeldrapper

### CODE REVIEW

File: actionpack/lib/action_controller/metal/implicit_render.rb [hunk 1]
```diff
@@ -35,7 +35,9 @@ module ImplicitRender
     include BasicImplicitRender
 
     def default_render
-      if template_exists?(action_name.to_s, _prefixes, variants: request.variant)
+      if Object.const_defined?(const_name = renderable_name(action_name.to_s.underscore, _prefixes))
+        render(const_name.constantize.new)
```
Comment (MatheusRich) [inline actionpack/lib/action_controller/metal/implicit_render.rb:39]: I'm sorry if I'm missing something. How would you pass initialization arguments to a renderable?

**Comment by bradgessler** [comment]
Glad to see this in a Rails PR! I have a few thoughts after using class names for views in projects that are worth considering.

First, if a project has conventions like `Views::Posts::Show` in `./app/views/posts/show.rb`, I think it's helpful for devs to have to include the module namespace at the top of their controllers like this:

```rb
class PostsController < ApplicationController
  include Views::Posts
end
```

This is useful because highly-compenitized apps tend to re-use a lot of views. For example:

```rb
module Blog
  class PostsController < ApplicationController
     include Views::Posts
  end
end
```

The controller then just has to worry about finding the corresponding class in the controller, like I do at https://github.com/rubymonolith/superview/blob/main/lib/superview/actions.rb#L42

To the point @MatheusRich is making; assigning is an interesting problem. The approach I took in Superview is to assign `@foo` to `Component#foo=`. I avoided assigning directly to the instance variables inside the class because it felt like it broke encapsulation.

Finally, I do recommend breaking out at least two functions, one that resolves the class and the other that assigns variables, so that its easy for people to override for whatever framework, component system, or style they're using.

The module at https://github.com/rubymonolith/superview/blob/main/lib/superview/actions.rb might serve as useful prior art. if it makes more sense to merge that into Rails I could open a PR with tests, but I don't want to spend time doing that unless some Rails core folks express interest in merging something like this.

File: actionpack/lib/action_controller/metal/implicit_render.rb (hunk 1 above)
Comment (joelhawksley) [reply actionpack/lib/action_controller/metal/implicit_render.rb:39]: @MatheusRich you aren't missing something. In this implementation, arguments are not passed to a renderable when it is rendered implicitly. How do you feel about @bradgessler's proposed solution (with prior working art) to assign ivars to renderables?

**Comment by joelhawksley** [comment]
> I think it's helpful for devs to have to include the module namespace at the top of their controllers like this:

@bradgessler my gut says that we should have implicit rendering Just Work ™️ and not require (hah) an explicit include. It just feels more _Railsy_ to me that way.

On a similar topic: I'm also wondering if we should instead use a `View` suffix instead of a module prefix. I like the idea of having Class names that include the type, as Rails already does for everything except for models. I'm going to go ahead and update this branch to use that convention.

File: actionpack/lib/action_controller/metal/implicit_render.rb (hunk 1 above)
Comment (MatheusRich) [reply actionpack/lib/action_controller/metal/implicit_render.rb:39]: I like the idea of having a method that automatically assigns ivars but can be overridden.

**Comment by dhh** [comment]
Appreciate this exploration, but I'm not a fan of a convention that requires a deeply nested namespace to connect here. I think this is better pursued in a gem that can encapsulate all there is around a given view model convention.

**Pull request closed without merging**